### PR TITLE
Add cluster lifecycle information to docs

### PR DESCRIPTION
**Pull request by chrisnegus**
*Issue:* [#5119](https://github.com/aws/eks-anywhere/issues/5119)
*Description of changes:* This PR starts the documentation process for adding cluster lifecycle features to EKS Anywhere docs. In particular, it notes the ability to use kubectl to create workload clusters for vSphere, Docker, and Snow. It also describes this feature in the cluster workflow and cluster topology docs.

I still have some questions, so I have not yet added cluster lifecycle content for Nutanix, Bare Metal, or Cloudstack. I also need more information on how cluster lifecycle should be described in relation to workload cluster upgrades and deletions. I will either add that information here or create new PRs after this one merges.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/concepts/cluster-topologies.md [hunk 1]
```diff
@@ -33,6 +33,15 @@ This shows examples of a management cluster that deploys and manages multiple wo
...
 ![Management clusters can create and manage multiple workload clusters](/images/eks-a_cluster_management.png)
 
+With the management cluster in place, you have a choice of tools for creating, upgrading, and deleting workload clusters.
+Check each provider to see which tools it currently supports.
+Supported workload cluster creation tools include:
+
+* `eksctl` CLI
+* Terraform
+* GitOps
+* `kubectl` CLI or similar tools that can communicate with the Kubernetes API
```
Comment (ddjjia) [inline docs/content/en/docs/concepts/cluster-topologies.md:43]: what do you mean similar tool here?

File: docs/content/en/docs/getting-started/local-environment/_index.md [hunk 2]
```diff
@@ -323,6 +323,11 @@ Follow these steps to have your management cluster create and manage separate wo
    * **GitOps**: See [Manage separate workload clusters with GitOps]({{< relref "../../tasks/cluster/cluster-flux.md#manage-separate-workload-clusters-using-gitops" >}})
 
    * **Terraform**: See [Manage separate workload clusters with Terraform]({{< relref "../../tasks/cluster/cluster-terraform.md#manage-separate-workload-clusters-using-terraform" >}})
+
+   * **kubectl CLI**: The cluster lifecycle feature lets you use `kubectl`, or other tools that that can talk to the Kubernetes API, to create a workload cluster. To use `kubectl`, run:
+      ```bash
+      kubectl apply -f eksa-w01-cluster.yaml 
+      ```
```
Comment (ddjjia) [inline docs/content/en/docs/getting-started/local-environment/_index.md:330]:
```suggestion
   * **kubectl CLI**: The cluster lifecycle feature lets you use `kubectl`, or other tools that that can talk to the Kubernetes API, to create a workload cluster. To use `kubectl`, run:
      ```bash
      kubectl apply -f eksa-w01-cluster.yaml 
      ```
```
   * **Kubernetes CLI**: The cluster full lifecycle feature lets you use `kubectl` to manage a workload cluster. Example:
      ```bash
      kubectl apply -f eksa-w01-cluster.yaml 
      ```

File: docs/content/en/docs/concepts/cluster-topologies.md (hunk 1 above)
Comment (chrisnegus) [reply docs/content/en/docs/concepts/cluster-topologies.md:43]: I changed it to just: `kubectl` CLI to communicate with the Kubernetes API

File: docs/content/en/docs/concepts/clusterworkflow.md [hunk 3]
```diff
@@ -8,8 +8,9 @@ description: >
 ---
 
 Each EKS Anywhere cluster is built from a cluster specification file, with the structure of the configuration file based on the target provider for the cluster.
-Currently, Bare Metal, CloudStack, and VMware vSphere are the recommended providers for supported EKS Anywhere clusters.
-We step through the cluster creation workflow for those providers here.
+Currently, Bare Metal, CloudStack, Nutanix, Snow, and VMware vSphere are the recommended providers for supported EKS Anywhere clusters.
+Docker is available as an unsupported provider.
```
Comment (ddjjia) [inline docs/content/en/docs/concepts/clusterworkflow.md:12]: We also support docker

File: docs/content/en/docs/concepts/cluster-topologies.md [hunk 4]
```diff
@@ -33,6 +33,15 @@ This shows examples of a management cluster that deploys and manages multiple wo
 
 ![Management clusters can create and manage multiple workload clusters](/images/eks-a_cluster_management.png)
 
+With the management cluster in place, you have a choice of tools for creating, upgrading, and deleting workload clusters.
+Check each provider to see which tools it currently supports.
+Supported workload cluster creation tools include:
```
Comment (jiayiwang7) [inline docs/content/en/docs/concepts/cluster-topologies.md:38]: should we mention upgrade and delete? :0
```suggestion
Supported workload cluster creation/upgrade/delete tools include:
```

File: docs/content/en/docs/concepts/clusterworkflow.md (hunk 3 above)
Comment (chrisnegus) [reply docs/content/en/docs/concepts/clusterworkflow.md:12]: I think the wording is correct: "Docker is available as an unsupported provider."

File: docs/content/en/docs/concepts/cluster-topologies.md (hunk 4 above)
Comment (chrisnegus) [reply docs/content/en/docs/concepts/cluster-topologies.md:38]: I'll add that wording here. But I'll look into adding information on upgrade and delete in other appropriate section of the docs in a separate PR.

**Pull request merged**